From 529d0055cd089b3b9c7b4bc513aa078ffee3d7a5 Mon Sep 17 00:00:00 2001
From: Ilya Ledvich <ilya@compulab.co.il>
Date: Tue, 15 Dec 2020 12:06:36 +0200
Subject: [PATCH 6/8] arch: arm64: imx8qm: cl-som-imx8max: enable DSI-to-HDMI
 bridge

Enable DV7535 MIPI DSI to HDMI bridge on MIPI1_DSI interface.
Communication with the bridge is performed via MIPI_DSI1_I2C0 bus.

Signed-off-by: Ilya Ledvich <ilya@compulab.co.il>
---
 .../dts/freescale/imx8qm-cl-som-imx8max.dtsi  |  7 +++
 .../boot/dts/freescale/imx8qm-sbc-imx8max.dts | 48 +++++++++++++++++++
 2 files changed, 55 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/imx8qm-cl-som-imx8max.dtsi b/arch/arm64/boot/dts/freescale/imx8qm-cl-som-imx8max.dtsi
index eb3ad4eebb06..918e0bb9f17f 100755
--- a/arch/arm64/boot/dts/freescale/imx8qm-cl-som-imx8max.dtsi
+++ b/arch/arm64/boot/dts/freescale/imx8qm-cl-som-imx8max.dtsi
@@ -698,6 +698,13 @@
 		>;
 	};
 
+	pinctrl_mipi1_lpi2c0: mipi1_lpi2c0grp {
+		fsl,pins = <
+			IMX8QM_MIPI_DSI1_I2C0_SCL_MIPI_DSI1_I2C0_SCL		0xc6000020
+			IMX8QM_MIPI_DSI1_I2C0_SDA_MIPI_DSI1_I2C0_SDA		0xc6000020
+		>;
+	};
+
 	pinctrl_lpuart0: lpuart0grp {
 		fsl,pins = <
 			IMX8QM_UART0_RX_DMA_UART0_RX				0x06000020
diff --git a/arch/arm64/boot/dts/freescale/imx8qm-sbc-imx8max.dts b/arch/arm64/boot/dts/freescale/imx8qm-sbc-imx8max.dts
index 77e093af1962..56c999a121dc 100755
--- a/arch/arm64/boot/dts/freescale/imx8qm-sbc-imx8max.dts
+++ b/arch/arm64/boot/dts/freescale/imx8qm-sbc-imx8max.dts
@@ -127,6 +127,53 @@
 	};
 };
 
+&i2c0_mipi1 {
+	#address-cells = <1>;
+	#size-cells = <0>;
+        pinctrl-names = "default";
+        pinctrl-0 = <&pinctrl_mipi1_lpi2c0>;
+	clock-frequency = <100000>;
+	status = "okay";
+
+	adv_bridge1: adv7535@3d {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		compatible = "adi,adv7535";
+		reg = <0x3d>;
+		adi,addr-cec = <0x3b>;
+		adi,dsi-lanes = <4>;
+		adi,dsi-channel = <1>;
+		interrupt-parent = <&lsio_gpio1>;
+		interrupts = <23 IRQ_TYPE_LEVEL_LOW>;
+		status = "okay";
+
+		port@0 {
+			reg = <0>;
+			adv7535_1_in: endpoint {
+				remote-endpoint = <&mipi1_adv_out>;
+			};
+		};
+	};
+};
+
+&mipi1_dphy {
+	status = "okay";
+};
+
+&mipi1_dsi_host {
+	status = "okay";
+
+	ports {
+		port@1 {
+			reg = <1>;
+			mipi1_adv_out: endpoint {
+				remote-endpoint = <&adv7535_1_in>;
+			};
+		};
+	};
+};
+
 &lsio_gpio1 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_lsio_gpio1>;
@@ -166,6 +213,7 @@
 	pinctrl_lsio_gpio1: lsiogpio1grp {
 		fsl,pins = <
 			IMX8QM_MIPI_DSI0_GPIO0_00_LSIO_GPIO1_IO18		0x06000020 /* MIPI_DSI0_PWM */
+			IMX8QM_MIPI_DSI1_GPIO0_01_LSIO_GPIO1_IO23		0x00000020 /* MIPI_DSI1_INT */
 		>;
 	};
 
-- 
2.20.1

